### PR TITLE
implementation of general nuclear potential

### PR DESCRIPTION
**Pull request by Christian48596**

### CODE REVIEW

File: one_electron.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    compute_last_energy = False
+    
+    P = vp.PoissonOperator(mra, prec)
+    light_speed = spinorb1.light_speed
+
+    while (error_norm > prec or compute_last_energy):
+        n_22 = spinorb1.overlap_density(spinorb1, prec)
+
+        # Definition of two electron operators
+        B22    = P(n_22.real) * (4 * np.pi)
```
Comment (ilfreddy) [inline one_electron.py:25]: The names are still reminescent of the 2e code. There is no 2nd electron here :-)

File: one_electron.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        print("mu: ", mu)
+        
+
+        new_spinorb1 = orb.apply_helmholtz(RHS, mu, prec)
+        print("normalization")
+        new_spinorb1.normalize()
+        print("crop")
+        new_spinorb1.cropLargeSmall(prec)
+        new_spinorb1.append(new_spinorb1)
+        new_spinorb1.append(new_spinorb1.ktrs(prec))
```
Comment (ilfreddy) [inline one_electron.py:119]: There is only one orbital. No need to use ktrs, nor to "append" the original orbital to a list.

File: one_electron.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        mu = orb.calc_non_rel_mu(cke+cpe)
+        print("mu: ", mu)
+        
+
+        new_spinorb1 = orb.apply_helmholtz(RHS, mu, prec)
+        print("normalization")
+        new_spinorb1.normalize()
+        print("crop")
+        new_spinorb1.cropLargeSmall(prec)
+        new_spinorb1.append(new_spinorb1)
```
Comment (ilfreddy) [inline one_electron.py:118]: This circular reference is for sure not going to work...

File: test.py [hunk 4]
```diff
@@ -90,26 +119,45 @@
...
-            Peps = vp.ScalingProjector(mra,prec/10)
-            f = lambda x: nucpot.homogeneus_charge_sphere(x, origin, Z, atom)
-            V_tree = Peps(f)
-        elif args.potential == 'gaussian':
-            Peps = vp.ScalingProjector(mra,prec/10)
-            f = lambda x: nucpot.gaussian(x, origin, Z, atom)
-            V_tree = Peps(f)
+        V_tot = vp.FunctionTree(mra)
+        V_tot.setZero()
+        for atom, origin in coordinates.items():
```
Comment (ilfreddy) [inline test.py:129]: All this stuff should go in the nuclear potential file. Here you just want to (1) parse the gepmetry (2) send it to the appropriate function which will return the nuclear potential

File: test.py [hunk 5]
```diff
@@ -137,18 +185,24 @@
 
     length = 2 * args.box
 
+    if runD_1e:
```
Comment (ilfreddy) [inline test.py:188]: This does not match the boolean you have above.

File: test.py [hunk 6]
```diff
@@ -137,18 +185,24 @@
...
+        spinorb1 = one_electron.gs_D_1e(spinorb1, V_tree, mra, prec)
+
+    if runD2_1e:
```
Comment (ilfreddy) [inline test.py:191]: This does not match the boolean you have above.

File: one_electron.py (hunk 1 above)
Comment (Christian48596) [reply one_electron.py:25]: Changed into n_11 and B11

File: one_electron.py (hunk 2 above)
Comment (Christian48596) [reply one_electron.py:119]: OK

**Review by ilfreddy — CHANGES_REQUESTED** [review]

File: atom_list.txt
[the remark this answers is not in the document]
Comment (Christian48596) [reply atom_list.txt:1]: ok

**Review by ilfreddy — APPROVED** [review]

**Pull request merged**